src/analysis/processing/qgsalgorithmdensifygeometriesbycount.cpp

```diff
@@ -0,0 +1,124 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgsalgorithmdensifygeometriesbycount.h"
+
+///@cond PRIVATE
+
+QString QgsDensifyGeometriesByCountAlgorithm::name() const
+{
+  return QStringLiteral( "densifygeometries" );
+}
+
+QString QgsDensifyGeometriesByCountAlgorithm::displayName() const
+{
+  return QObject::tr( "Densify by count" );
+}
+
+QStringList QgsDensifyGeometriesByCountAlgorithm::tags() const
+{
+  return QObject::tr( "add,vertex,vertices,points,nodes" ).split( ',' );
+}
+
+QString QgsDensifyGeometriesByCountAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsDensifyGeometriesByCountAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsDensifyGeometriesByCountAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm takes a polygon or line layer"
+                      "and generates a new one in which the geometries"
+                      "have a larger number of vertices than the "
+                      "original one. If the geometries have z or m values"
+                      "present then these will be linearly interpolated"
+                      "at the added nodes. The number of new vertices to"
+                      "add to each feature geometry is specified "
+                      "as an input parameter." );
+}
+
+QString QgsDensifyGeometriesByCountAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Creates a densified version of geometries." );
+}
+
+QgsDensifyGeometriesByCountAlgorithm *QgsDensifyGeometriesByCountAlgorithm::createInstance() const
+{
+  return new QgsDensifyGeometriesByCountAlgorithm;
+
+}
+
+QList<int> QgsDensifyGeometriesByCountAlgorithm::inputLayerTypes() const
+{
+  return QList<int>() << QgsProcessing::TypeVectorLine << QgsProcessing::TypeVectorPolygon;
+}
+
+void QgsDensifyGeometriesByCountAlgorithm::initParameters( const QVariantMap &configuration )
+{
+  Q_UNUSED( configuration )
+  std::unique_ptr<QgsProcessingParameterNumber> verticesCnt = qgis::make_unique<QgsProcessingParameterNumber>( QStringLiteral( "VERTICES" ),
+      QObject::tr( "Number of vertices to add" ),
+      QgsProcessingParameterNumber::Integer,
+      1, false, 1, 10000000 );
+  verticesCnt->setIsDynamic( true );
+  verticesCnt->setDynamicPropertyDefinition( QgsPropertyDefinition( QStringLiteral( "VerticesCount" ), QObject::tr( "Vertices count" ), QgsPropertyDefinition::IntegerPositiveGreaterZero ) );
```

Comment:
```suggestion
  verticesCnt->setDynamicPropertyDefinition( QgsPropertyDefinition( QStringLiteral( "VerticesCount" ), QObject::tr( "Vertices count" ), QgsPropertyDefinition::IntegerPositive ) );
```

I think we should allow zero counts here -- e.g. if you only want to densify lines greater than a certain threshold